luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/AbridgedLogicalTableGroupFactory.java

```diff
@@ -0,0 +1,103 @@
+ * This concept exists for the reason that when we configure LogicalTable,
+ * each granularity will be treated as a separated LogicalTable internally.
+ * Thus one configuration correspond to a LogicalTableGroup, and the infrastructure
+ * will support look up for each individual LogicalTable with a TableIdentifier.
+ */
+public class AbridgedLogicalTableGroupFactory implements Factory<LogicalTableGroup> {
+    private static final String ENTITY_TYPE = "abridged logical table group";
+
+    /**
+     * Build a group of LogicalTable, it has one LogicalTable per granularity.
+     *
+     * @param name  apiName of the LogicalTable
+     * @param configTable  the LogicalTableGroup's configuration
+     * @param resourceFactories  the source for locating dependent objects
+     *
+     * @return  A newly constructed config instance for the name and config provided
+     */
+    @Override
+    public LogicalTableGroup build(String name, ObjectNode configTable, LuthierIndustrialPark resourceFactories) {
+        validateFields(name, configTable);
+
+        LogicalTableGroup tableDictionary = new LogicalTableGroup();
+        LogicalTableName tableName = LogicalTableName.forName(name);
+        MetricDictionary metricDictionary = resourceFactories.getMetricDictionary();
+        LinkedHashSet<Dimension> dimensions = new LinkedHashSet<>();
+        LinkedHashSet<PhysicalTable> physicalTables = new LinkedHashSet<>();
+        LinkedHashSet<ApiMetricName> apiMetricNames = new LinkedHashSet<>();
+        configTable.get("dimensions").forEach(
+                node -> dimensions.add(resourceFactories.getDimension(node.textValue()))
+        );
+        configTable.get("physicalTables").forEach(
+                node -> physicalTables.add(resourceFactories.getPhysicalTable(node.textValue()))
+        );
+        configTable.get("metrics").forEach(
+                node -> {
+                    /* go to the LIP to retrieve the metric */
+                    /* Must be done after having actual metric builders! */
+                    // apiMetricNames.add(new LuthierApiMetricName(node.textValue(), granularity));
+                    // metricDictionary.add();
+                }
+        );
+        TableGroup tableGroup = new TableGroup(
+                physicalTables,
+                apiMetricNames,
+                dimensions
+        );
+
+        configTable.get("granularities").forEach(
+                (JsonNode node) -> {
+                    ZonedTimeGrain granularity = DefaultTimeGrain.valueOf(
+                            node.textValue().toUpperCase(Locale.US)
+                    ).buildZonedTimeGrain(DateTimeZone.forID(configTable.get("dateTimeZone").textValue()));
+                    LogicalTable logicalTable = new LogicalTable(tableName, granularity, tableGroup, metricDictionary);
+                    tableDictionary.put(new TableIdentifier(logicalTable), logicalTable);
```

Comment:
Use the "    public LogicalTable(
            @NotNull String name,
            String category,
            String longName,
            @NotNull Granularity granularity,
            ReadablePeriod retention,
            String description,
            TableGroup tableGroup,
            MetricDictionary metricDictionary"
constructor

Reply:
Implement retention using joda.time.Period.parse

Reply:
Don't bother with the LogicalTableName constructor, it's just busywork here.

Reply:
sure